### PR TITLE
[1.5] bump k8s.io dependencies to v0.26.4

### PR DESCRIPTION
**Pull request by liangyuanpeng**
What type of PR is this?

/kind cleanup

What this PR does / why we need it:

Clients using client-go v0.26.0 - v0.26.3 can panic under the following conditions
- API Server v1.27+ or v1.26 and feature gate enabled --feature-gates=AggregatedDiscoveryEndpoint=true
- Cluster contains aggregated api server for custom metrics (e.g. Prometheus) or any aggregated api server returning malformed discovery.

Ref: https://github.com/kubernetes/kubernetes/issues/119840

Related:
- https://github.com/karmada-io/karmada/issues/3924

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Bump k8s.io dependencies to v0.26.4
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -21,27 +21,27 @@ require (
...
-	k8s.io/apiextensions-apiserver v0.26.1
-	k8s.io/apimachinery v0.26.1
-	k8s.io/apiserver v0.26.1
-	k8s.io/cli-runtime v0.26.1
-	k8s.io/client-go v0.26.1
-	k8s.io/cluster-bootstrap v0.26.1
-	k8s.io/code-generator v0.26.1
-	k8s.io/component-base v0.26.1
-	k8s.io/component-helpers v0.26.1
+	k8s.io/api v0.26.2
```
Comment (XiShanYongYe-Chang) [inline go.mod:31]: I understand that I should upgrade to v0.26.4, right?

Comment (liangyuanpeng) [reply go.mod]: Seems like there have some confusion in my workspace. Updated,Thanks!

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks
/lgtm

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**